### DISCUSSION TITLE
Don't appear in action list to add

### DISCUSSION THREAD

**Discussion by julienfollow** [dasniko/keycloak-2fa-sms-authenticator#27] [Q&A]
Hi
I want try this extension on local environment to test it. I want to be inspired to develop a similar module for implement twilio services. 
But, when a build it with `mvn clean package` and after copy/paste `.jar `on my docker container in `/opt/keycloak/standalone/deployments`, the action doesn't appear in action list on authentication flow. 

I have no error in my log about this issue. 
![image](https://user-images.githubusercontent.com/70938205/189358231-e5475053-050b-483c-8147-d39a14c48eb1.png)

ps: when i build my custom docker image, i need to create the deployments folder because don't exist on default keycloak image `FROM quay.io/keycloak/keycloak:19.0.0`

Thanks

**Comment by dasniko** [comment]
Things have changed a lot in Keycloak in the last few months. Please refer to the official documentation on how to deploy custom providers, the directory has changed to `/opt/keycloak/providers`.
Additionally, KC19 has a bug which prevents this extension to work properly.

**Reply by robertcsakany** [reply]
Hi! I have the very same issue with 21.0.2. I've rebuilt the plugin with this keycloak version  and the plugin deployed to the correct deploy folder. In the log there is no error message, it seems OK, but we have this very same issue. Can you tell about that KC19 issue, is there any ticket which relates it? Thank you.

**Reply by julienfollow** [reply]
Hi robert, do you want use this library to create a custom 2fa by sms ?

**Reply by dasniko** [reply]
Works here without problems. 🤷‍♂️
